python_modules/dagster/dagster_tests/declarative_automation_tests/daemon_tests/test_asset_daemon.py

```diff
@@ -108,8 +112,38 @@ def _get_threadpool_executor(instance: DagsterInstance):
         yield executor
 
 
+mid_iteration_terminate_scenario = AssetDaemonScenario(
+    id="two_distinct_graphs_so_multiple_runs",
+    initial_spec=two_distinct_partitions_graphs.with_current_time(time_partitions_start_str)
+    .with_current_time_advanced(days=1, hours=1)
+    .with_all_eager(),
+    execution_fn=lambda state: state.with_runs(
+        *[
+            run_request(["A"], partition_key=hour_partition_key(state.current_time, delta=-i))
+            for i in range(25)
+        ],
+        run_request(["C"], partition_key=day_partition_key(state.current_time)),
+    )
+    .evaluate_tick(stop_mid_iteration=True)
+    .assert_requested_runs_for_stopped_iteration(
+        1,
+        run_request(asset_keys=["B"], partition_key=hour_partition_key(state.current_time)),
+        run_request(asset_keys=["D"], partition_key=day_partition_key(state.current_time)),
+    )
+    .with_current_time_advanced(minutes=10)
+    .evaluate_tick()
+    .assert_requested_runs()  # cursor was updated by the last tick, so we don't have new runs
```

Comment:
TODO - in the sensor version of this test, no runs are requested on the second evaluation, but in the non-sensor tests, the run request for the asset that wasn't materialized in the first evaluation is requested. Might just be that the cursors are different, but could also be a bug

Reply:
huh, i'm surprised that would differ depending on whether its a sensor or not..

Reply:
@OwenKephart I don't think i understand AMP/DA enough to know if this is expected or indicates a bug in the implementation of stopping the tick